spec/pdfs/siren_infos_pdf_spec.rb

```diff
@@ -148,4 +148,28 @@
       its(:size) { is_expected.to eq 15 }
     end
   end
+
+  describe 'production issue' do
+    let(:dossier) { DossierEntreprise.new.tap(&:save) }
+
+    before do
+      create :representant, dossier_entreprise: dossier, type_representant: 'P. Physique', nom_patronyme: 'BUGGY', prenoms: 'spacing'
+    end
+
+    it { is_expected.to include('Nom prénoms: BUGGY spacing') }
+  end
```

Comment:
Le test est clean, mais manque de clarté : il y a un unique example dans le groupe 'production issue' (qui en soit n'aide pas à comprendre ce qui est testé) donc le hook `before` est overkill, et je pense que le groupe `describe` aussi.

J'écrirai le test comme ça : 

```ruby 
it 'writes representant identity even when the type "P. Physique" contains a space' do 
  malformed_type = 'P. Physique'  
  create :representant, type_representant: malformed_type, ...

  expect(subject).to include('Nom prénoms: BUGGY spacing')
end 
```

Je pense que les one liner avec it/its sont à éviter dans 90% des cas ; c'est très bien pour genre tester toutes les clés d'une payload JSON, tu sets le subject une fois et tu testes tout le temps la même chose donc l'objectif est évident, là je suis même pas certain que tu gagnes en nombre de ligne de code en comparaison à un unique bloc `it` plus verbeux, par contre tu perds en clarté du comportement testé.

Reply:
j'ai modifié pour proposer une autre solution